docs/sources/mimir/query/query-best-practices.md

```diff
@@ -0,0 +1,84 @@
+---
+description: Learn best practices for querying Grafana Mimir.
+title: Best practices for querying Grafana Mimir
+menuTitle: Query best practices
+weight: 1
+keywords:
+  - query
+  - best practices
+  - performance
+---
+
+<!-- Note: This topic is mounted in the GEM documentation. Ensure that all updates are also applicable to GEM. -->
+
+# Best practices for querying Grafana Mimir
+
+The way you write queries for Grafana Mimir affects the speed and quality of your results. Follow these best practices to ensure optimal performance and reliability for queries.
+
+## Use precise label selectors
+
+To reduce the initial dataset size, start with the most specific label selectors. For example, if you have labels like `namespace` and `app_name` where `app_name` is more specific, lead with `app_name="myapp"` rather than starting with the broader `namespace` selector.
+
+To further reduce the number of series that require processing, include as many label selectors as possible for your query.
+
+Whenever possible, use an exact label match, such as `label="value"`, instead of a regular expression, such as `label=~"pattern"`. Regular expressions are more computationally expensive.
+
+For more information about how to query metric labels, refer to [Query metric labels](https://grafana.com/docs/mimir/<MIMIR_VERSION>/query/query-metric-labels/).
+
+## Narrow down your time range
+
+Limit your query to a specific time period to reduce the number of metrics Mimir needs to process. As a best practice, query the shortest feasible time range to narrow down your results. Larger time ranges require more computing resources and increase query latency.
+
+Follow these guidelines for setting a time range based on your query type:
+
+- Dashboards: For dashboards, use a time range that matches the typical viewing window. While users can zoom out to view more history, the default view should be efficient and focused on recent, relevant data.
+- Alerting rules: Alerting rules should almost always operate on short, recent time windows to detect current states or recent trends, rather than long-term historical patterns. Timely alerts depend on the quick evaluations of fresh data.
+- Ad-hoc analysis: When exploring data, start with small time ranges and gradually expand if more historical context is required for your investigation. This iterative approach prevents unnecessary strain on the system during exploratory phases.
+
+If you don't specify a time range in your query, consider using the `-store.max-labels-query-length` CLI flag to limit the maximum time range.
```

Comment:
Specifying a time range is always required for range queries, and defaults to the current time for instant queries, which seem to be the types of queries we're talking about here. The `-querier.max-query-lookback` and `-querier.max-partial-query-length` CLI flags limit how far back and how long a query's time range can be.